src/mapgen_functions.cpp

```diff
@@ -899,27 +899,50 @@ void mapgen_road( mapgendata &dat )
         }
 
         // four-way intersections
-        if( num_dirs == 4 && one_in( 2 ) ) {
-            m->furn_set( point( 3, 3 ), f_traffic_light );
-            m->furn_set( point( 3, 20 ), f_traffic_light );
-            m->furn_set( point( 20, 3 ), f_traffic_light );
-            m->furn_set( point( 20, 20 ), f_traffic_light );
+        if( num_dirs == 4 ) {
+            square( m, t_zebra, point( 4, 1 ), point( 19, 2 ) );
```

Comment:
```suggestion
            // No need to draw a zebra if there is no sidewalk nearby
            if (m->ter(point(3, 1)) == t_sidewalk && m->ter(point(20, 2)) == t_sidewalk) {
                square( m, t_zebra, point( 4, 1 ), point( 19, 2 ) );
            }
```
It should be copied for other 3 cases.

Reply:
Also I'e saw google maps. Not every T intersection has all pedestrian crossings. And not every 4 way intersection has 2 crossings. Also, in some suburbs there are almost no pedestrian crossings. For example 1 for 5-10 intersections.

Reply:
I guess adding some randomness in placing crossings wouldn't hurt. Will get on this tomorrow.